### ISSUE TITLE
Add checksum for downloaded archives

### ISSUE THREAD

**Issue by mirkobrombin** [Vanilla-OS/Vib#2]

**Comment by arejula27** [comment]
I would like to contribute, can you describe the issue ?

**Comment by mirkobrombin** [comment]
For every non-git source, we should define its hash so that Vib can compare that with the downloaded one

**Comment by arejula27** [comment]
The checksum should be added to the type Source struct {} and then compared with the resource downloaded in the function DownloadTarSource?
```go
func DownloadTarSource(recipe *Recipe, source Source) error {}
```

**Comment by mirkobrombin** [comment]
Yup, but be sure to exclude git sources, only archives should be checked

**Comment by arejula27** [comment]
So it is only required on `DownloadTarSource` or should I also check something on make, meson, etc. types? 
As I understand in that function there are not git sources, right?
